examples/webgl_materials_envmaps_hdr.html

```diff
@@ -215,12 +219,53 @@
 
 				window.addEventListener( 'resize', onWindowResize, false );
 
+				function updateEnvMap() {
+
+					var newEnvMap = standardMaterial.envMap;
+
+					switch( params.envMap ) {
+
+						case 'None': newEnvMap = null; break;
+						case 'LDR': newEnvMap = ldrCubeRenderTarget.texture; break;
+						case 'HDR': newEnvMap = hdrCubeRenderTarget.texture; break;
+						case 'RGBM16': newEnvMap = rgbmCubeRenderTarget.texture; break;
+
+					}
+
+					if( newEnvMap !== standardMaterial.envMap ) {
+
+						standardMaterial.envMap = newEnvMap;
+						standardMaterial.needsUpdate = true;
+						floorMaterial.emissive = new THREE.Color( 1, 1, 1 );
+						floorMaterial.emissiveMap = newEnvMap;
+						floorMaterial.needsUpdate = true;
+
+					}
+
+				}
+
 				var gui = new dat.GUI();
 
-				gui.add( params, 'envMap', [ 'None', 'LDR', 'HDR', 'RGBM16' ] );
-				gui.add( params, 'roughness', 0, 1 );
-				gui.add( params, 'bumpScale', - 1, 1 );
-				gui.add( params, 'exposure', 0.1, 2 );
+				gui.add( params, 'envMap', [ 'None', 'LDR', 'HDR', 'RGBM16' ] ).onChange( updateEnvMap ).listen();
+
+				gui.add( params, 'roughness').min( 0 ).max( 1 ).onChange( function() {
+
+					standardMaterial.roughness = params.roughness;
+
+				} );
+
+				gui.add( params, 'bumpScale').min( - 1 ).max( 1 ).onChange( function() {
+
+					standardMaterial.bumpScale = - 0.05 * params.bumpScale;
+
+				} );
+
+				gui.add( params, 'exposure').min( 0.1 ).max( 2 ).onChange( function() {
+
+					renderer.toneMappingExposure = Math.pow( params.exposure, 4.0 );
+
+				} );
```

Comment:
wouldn't do that - three saved assignments once a frame won't justify the callbacks and extra size.